flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManagerDriver.java

```diff
@@ -228,6 +228,8 @@ public CompletableFuture<KubernetesWorkerNode> requestResource(
                                             podName);
                                     stopPod(taskManagerPod.getName());
                                 }
+                            } else if (t instanceof RetryableException) {
```

Comment:
That should work. Wouldn't we achieve the same thing by cancelling the request instead of completing the it exceptionally and ensuring the `stopPod` operation idempotent? That way, we wouldn't add another code branch :thinking: I'm just curious about your opinion on that one.

Reply:
I think the semantics for the future being completed exceptionally and canceled are slightly different. Canceling means the resource is no longer needed, while completed exceptionally means the resource is still needed but for some reason it's not allocated successfully. That leads to different actions when handle the future in `ActiveResourceManager#requestNewWorker`.

Reply:
Makes sense. I missed that it's also handled in `ActiveResourceManager`. Thanks for clarification :-)